src/Perlang.Interpreter/Typing/TypeResolver.cs

```diff
@@ -126,18 +126,48 @@ public override VoidObject VisitBinaryExpr(Expr.Binary expr)
                     return VoidObject.Void;
 
                 case TokenType.STAR_STAR:
-                    double? leftMaxValue = GetApproximateMaxValue(leftTypeReference.ClrType);
-                    double? rightMaxValue = GetApproximateMaxValue(rightTypeReference.ClrType);
-
-                    if (leftMaxValue == null || rightMaxValue == null)
+                    // TODO: If expr.Left is an Expr.Identifier, it might refer to a variable (= supported left-hand
+                    // TODO: operand), a function call (also supported) _or_ a function/method reference. This is _not_
+                    // TODO: supported at this stage. I think we should somehow work on making it possible to
+                    // TODO: distinguish between `foo` and `foo()` - perhaps the inferred type of the former should be
+                    // TODO: something like `PerlangFunction` instead of the return type of the function? This is
+                    // TODO: probably a significant change so be prepared that it will take some time.
+
+                    // Only a certain set of type combinations are supported with this operator. We validate it here to
+                    // be able to provide "compile-time" error checking rather than runtime checking.
+
+                    if ((leftTypeReference.ClrType == typeof(int) || leftTypeReference.ClrType == typeof(long)) &&
+                        rightTypeReference.ClrType == typeof(int))
+                    {
+                        // TODO: This is not 100% sane. The return type depends on the _value_ of the right-hand
+                        // TODO: expression in this case. (See the STAR_STAR implementation in VisitBinaryExpr() for the full
+                        // TODO: details). So it can sometimes be a BigInteger and sometimes a double. This is insane
+                        // TODO: semantics, maybe we should just be consistent and make it always a double in this case...?
```

Comment:
We should perhaps look into this a bit more before merging.

Reply:
Decided to go with the "safer" approach in this case: negative exponents with `int`/`long`/`bigint` values will throw a runtime exception for now, and we retain the `bigint` return type for integer value operands.

At some point, it would be cool to pre-evaluate constant expressions at compile-stage, to be able to make negative constant exponents be a compile-time instead of runtime error. But... for Another Day. :blush: